controllers/agentpool_controller_deployment.go

```diff
@@ -224,7 +226,22 @@ func agentPoolDeploymentName(ap *appv1alpha2.AgentPool) string {
 }
 
 func agentPoolPodLabels(ap *appv1alpha2.AgentPool) map[string]string {
-	return map[string]string{
+	label := map[string]string{
 		poolNameLabel: ap.Name,
 	}
+
+	//Attempting to merge
+	if ap.Spec.AgentDeployment != nil && ap.Spec.AgentDeployment.Labels != nil {
```

Comment:
We can omit `ap.Spec.AgentDeployment != nil` since this validation has been done earlier. In other words, if we have reached this point, we are sure that `ap.Spec.AgentDeployment` is not `nil`.

Let's replace `ap.Spec.AgentDeployment.Labels != nil` with `len(ap.Spec.AgentDeployment.Labels) > 0`.

Reply:
Shall we move the built-in label assignment to the end of this function? In this case, even if a user configures this label, the controller will overwrite it.

Reply:
yup, since built in labels take precedence.